dl/src/main/scala/com/intel/analytics/bigdl/nn/JoinTable.scala

```diff
@@ -27,6 +27,11 @@ import scala.reflect.ClassTag
  * It is a table module which takes a table of Tensors as input and
  * outputs a Tensor by joining them together along the dimension `dimension`.
  *
+ * Note: the input tensor can have indefinite number of dimensions,
+ * so modules can not automatically inference whether the user use the batch or not
+ * and what the actual dimensions of input excluding the batch dimension.
+ * Hence `nInputDims` represents the actual dimensions of given input excluding the batch.
+ *
```

Comment:
This is somewhat confusing. Just say somehting like "the input to this layer is expected to a tensor, or a batch of tensors; when using mini-batch, a batch of tensors will be passed to the layer and the user need to specify the number of dimensions of each input tensor in the batch using nInputDims"